src/views/CreateCommentForm/CreateCommentForm.test.tsx

```diff
@@ -8,29 +8,97 @@ vi.mock('../../shared/api/commentsApi', () => ({
+    name = 'Comment name';
+    body = 'Comment body';
     (createComment as Mock).mockReturnValue(new Promise(() => null));
   });
   describe('when the user types the name and the body', () => {
-    it('should make attempt to create the POST request', () => {
-      const createCommentForm = render(<CreateCommentForm />);
+    describe('and clicks the submit button', () => {
+      it('should make attempt to create the POST request', () => {
+        const createCommentForm = render(<CreateCommentForm />);
 
-      const nameInput = createCommentForm.getByPlaceholderText('name');
-      const bodyInput = createCommentForm.getByPlaceholderText('body');
+        const nameInput = createCommentForm.getByPlaceholderText('name');
+        const bodyInput = createCommentForm.getByPlaceholderText('body');
 
-      const name = 'Comment name';
-      const body = 'Comment body';
+        fireEvent.change(nameInput, { target: { value: name } });
+        fireEvent.change(bodyInput, { target: { value: body } });
 
-      fireEvent.change(nameInput, { target: { value: name } });
-      fireEvent.change(bodyInput, { target: { value: body } });
+        const button = createCommentForm.getByRole('button');
 
-      const button = createCommentForm.getByRole('button');
+        fireEvent.click(button);
 
-      fireEvent.click(button);
+        expect(createComment).toBeCalledWith({
+          name,
+          body,
+        });
+      });
+      describe('if the API call is successful', () => {
+        beforeEach(() => {
+          (createComment as Mock).mockResolvedValue({ name, body, id: 501 });
+        });
+        it('should display the created comment and not display an error', async () => {
+          const createCommentForm = render(<CreateCommentForm />);
+
+          const nameInput = createCommentForm.getByPlaceholderText('name');
+          const bodyInput = createCommentForm.getByPlaceholderText('body');
+
+          fireEvent.change(nameInput, { target: { value: name } });
+          fireEvent.change(bodyInput, { target: { value: body } });
+
+          const button = createCommentForm.getByRole('button');
+
+          fireEvent.click(button);
+
+          const nameParagraph = await createCommentForm.findByText(name, {
+            selector: 'p',
+          });
+          const bodyParagraph = await createCommentForm.findByText(body, {
+            selector: 'p',
+          });
+          const errorParagraph =
+            createCommentForm.queryByTestId('comment-error');
+
+          expect(nameParagraph).toBeDefined();
+          expect(bodyParagraph).toBeDefined();
+          expect(errorParagraph).toBeNull();
+        });
+      });
+      describe('if the API call is unsuccessfull', () => {
+        beforeEach(() => {
+          (createComment as Mock).mockImplementation(() => {
+            throw new Error();
+          });
+        });
+        it('should display an error and not display the comment', async () => {
```

Comment:
- not displaying the created comment
- displaying an error
are two separate test cases and should have two separate its